lib/checkPullRequestTemplate.js

```diff
@@ -165,69 +169,140 @@ const checkEssentialChecklist = (pullRequest) => {
         const isPrFromFork =
           pullRequest.head.repo.full_name === pullRequest.base.repo.full_name;
         if (!pullRequest.maintainer_can_modify && !isPrFromFork) {
-          if (message) {
-            message +=
-              'The allow edits from maintainers checklist needs to ' +
-              'be ticked so that maintainers can rerun failed tests. ' +
-              'Endeavour to add this by ticking on the check box. ';
-          } else {
-            message +=
-              'the allow edits from maintainers checklist needs to ' +
-              'be ticked so that maintainers can rerun failed tests. ' +
-              'Endeavour to add this by ticking on the check box. ';
-          }
+          message +=
+            'The allow edits from maintainers checklist needs to ' +
+            'be ticked so that maintainers can rerun failed tests. ' +
+            'Endeavour to add this by ticking on the check box. ';
         }
       }
     });
 
     return message;
   } else {
     return (
-      'the body of this PR is missing the checklist section, please ' +
+      'The body of this PR is missing the checklist section, please ' +
       'update it to include the checklist.'
     );
   }
 };
 
+
+/**
+ * This function checks if the pull request's body contains the
+ * proof of changes and returns an appropriate message if the required
+ * proofs are not provided.
+ *
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const checkProofThatChangesAreCorrect = (pullRequest) => {
+  /*
+    A valid Proof that changes are correct section would look like:
+    ## Proof that changes are correct
+
+    <!--
+    Add videos/screenshots of the user-facing interface to demonstrate
+    that the changes made in this PR work correctly.
+    -->
+    ![oppiabot-test](https://user-images.githubusercontent.com/30050862/
+      test.gif)
+    (or)
+    https://user-images.githubusercontent.com/30050862/
+      test.mp4
+      */
+  const proofOfChcangesHeadingIndex = pullRequest.body.indexOf(PROOF_TEXT);
+
+  if (proofOfChcangesHeadingIndex !== -1) {
```

Comment:
```suggestion
  if (proofOfChcangesHeadingIndex === -1) {
    return (
      'The body of this PR is missing the proof that changes ' +
      'are correct section, please update it to include the section.'
    );
  }
```

Reply:
And you can remove the code from below.

Reply:
Wouldn't be better to follow the same format as the other PR template checks, where the above comment would be returned only in the else section?

Reply:
I think that instead of having a big nested block it is better to keep the nested sections smaller.

Reply:
DONE